stdlib/asm/crypto/fri/frie2f4.masm

```diff
@@ -270,28 +338,152 @@ export.verify.4
     dup.2
     neq
 
+    # Save a word containing a fresh accumulator for Horner evaluating the remainder polynomial,
+    # a pointer to the evaluation point and a pointer to the location of the polynomial.
+    push.0.0 
+    exec.constants::tmp2 exec.constants::get_remainder_poly_address
+    exec.constants::tmp1 mem_storew
+    movup.4
+
     while.true
         # load [e0, e1, p, poe] from memory i.e. next query data (7 cycles)
-        push.0.0.0.0
         movup.4
         mem_loadw
         # => [poe, p, e1, e0, layer_ptr, rem_ptr, g, ...]
 
         # we now have everything to verify query p
-        exec.verify_query
+        exec.verify_query_64
 
         # prepare for next iteration (18 cycles)
         # => [x, x, x, x, x, x, x, x, x, x, g, ...]
-        dropw drop drop drop
+        dropw drop
         loc_loadw.0   # load [query_ptr, layer_ptr, rem_ptr, g]
         add.4
-        loc_storew.0  # store [query_ptr + 4, layer_ptr, rem_ptr, g]
-        dup
-        dup.2
+        loc_storew.0  # store [query_ptr + 1, layer_ptr, rem_ptr, g]
+        swapw
+        # => [x, x, x, x, query_ptr + 1, layer_ptr, rem_ptr, g, ...]
+        dup.5
+        dup.5
         neq
-        #=> [?, query_ptr + 4, layer_ptr, rem_ptr, g, ...]
+        #=> [?, query_ptr + 1, layer_ptr, rem_ptr, g, ...]
```

Comment:
Based on the comment on line 364, shouldn't this be `[?, x, x, x, x, query_ptr + 1, layer_ptr, rem_ptr, g, ...]`? I'm having trouble in general in this section to follow what's going on on the stack

Reply:
Agreed, there is a "garbage" word prepared for the `mem_loadw` in the next iteration.

Added a comment to help following stack transition